### PR TITLE
Pull Request for Issue2235: Add storage ring current to persistent view.

### PR DESCRIPTION
**Pull request by helfrij**
Added control editor to the persistent view for the storage ring current.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you take a look?

File: source/ui/BioXAS/BioXASPersistentView.cpp [hunk 1]
```diff
@@ -38,6 +38,13 @@ BioXASPersistentView::BioXASPersistentView(QWidget *parent) :
 
 	setLayout(layout);
 
+	// Create SR1 current view.
+
+	BioXASControlEditor *sr1CurrentEditor = new BioXASControlEditor(CLSStorageRing::storageRing()->ringCurrentControl());
+	sr1CurrentEditor->setTitle("SR1 Current");
```
Comment (XuetaoLiu-CLS) [inline source/ui/BioXAS/BioXASPersistentView.cpp]: the default value of `readOnly_` is false. Do we need to set it to true?

**Comment by XuetaoLiu-CLS** [comment]
Looks good other than a question on whether we should set the control to readonly.

Time Estimate: 5 minutes

File: source/ui/BioXAS/BioXASPersistentView.cpp (hunk 1 above)
Comment (dretrex) [inline source/ui/BioXAS/BioXASPersistentView.cpp]: Good point.

**Comment by dretrex** [comment]
Time Estimate: 1 minute

File: source/ui/BioXAS/BioXASPersistentView.cpp (hunk 1 above)
Comment (helfrij) [inline source/ui/BioXAS/BioXASPersistentView.cpp]: I checked, because I know we've had issues with this in the past. This control editor is read-only. :)

Comment (dretrex) [inline source/ui/BioXAS/BioXASPersistentView.cpp]: I can't seem to find where read-only is set to true.  Is there another mechanism that can set it to true?

Comment (helfrij) [inline source/ui/BioXAS/BioXASPersistentView.cpp]: I'm not sure there needs to be an explicit setting in this case. The ring current control is a read-only PV control in CLSStorageRing, and the control editor checks whether the control can move before displaying the 'edit dialog'. If the control can't move, the editor is effectively read-only. Is the worry that read-only PV controls may inappropriately report as being able to move?

**Pull request merged**